core/comment_events.js

```diff
@@ -96,14 +106,32 @@ Blockly.Events.CommentBase.prototype.toJson = function() {
 Blockly.Events.CommentBase.prototype.fromJson = function(json) {
   this.commentId = json['commentId'];
   this.group = json['group'];
+  this.blockId = json['blockId'];
+};
+
+/**
+ * Helper function for finding the comment this event pertains to.
+ * @return {?(Blockly.WorkspaceComment | Blockly.ScratchBlockComment)}
+ *     The comment this event pertains to, or null if it no longer exists.
+ * @private
+ */
+Blockly.Events.CommentBase.prototype.getComment_ = function() {
+  var workspace = this.getEventWorkspace_();
+  return workspace.getCommentById(this.commentId);
 };
 
 /**
  * Class for a comment change event.
  * @param {Blockly.WorkspaceComment} comment The comment that is being changed.
  *     Null for a blank event.
- * @param {string} oldContents Previous contents of the comment.
- * @param {string} newContents New contents of the comment.
+ * @param {!object} oldContents Object containing previous state of a comment's
```

Comment:
We should validate this, but doesn't have to be now.  I just filed an issue: #1565